secretmanager/api/SecretManagerTest/SamplesTest.cs

```diff
@@ -30,6 +30,7 @@ namespace GoogleCloudSamples
     public class SampleTests : IClassFixture<SecretsFixture>
     {
         private static readonly string s_projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        private static readonly string s_iamUser = Environment.GetEnvironmentVariable("DOTNET_SECRETMANAGER_SAMPLES_IAM_USER");
```

Comment:
Actually, so we don't have to set the environment variable, consider something like this:

```
var credential = ((ServiceAccountCredential)GoogleCredential.GetApplicationDefault().UnderlyingCredential);
string email = credential.Id;
```
This will give you the email address of the service account that executes the tests. Would that work for you?

Reply:
That will break for people who run tests locally though:

```text
Unable to cast object of type 'Google.Apis.Auth.OAuth2.UserCredential' to type 'Google.Apis.Auth.OAuth2.ServiceAccountCredential'.
```

IAM also requires that the entity is prefixed with the type (e.g. `serviceAccount:foo@bar.com`, `user:foo@bar.com`, `group:foo@bar.com`). I suppose we could add a big switch statement, but that feels like a lot of investment vs just making a static service account for these tets.

Reply:
Hmmm, true (I run locally with a service account but still might be someone who doesn't).
I'll set up the env variable then.

Reply:
Done